codebasin.py

```diff
@@ -120,25 +133,76 @@ def main():
     )
     rootdir = os.path.realpath(args.rootdir)
 
-    if args.config_file is None:
+    # Process the -p flag first to infer wider context.
+    filtered_platforms = []
+    additional_platforms = []
+    for p in args.platforms:
+        # If it's a path, it has to be a compilation database.
+        if os.path.exists(p):
+            if not os.path.splitext(p)[1] == ".json":
+                raise RuntimeError(f"Platform file {p} must end in .json.")
+            additional_platforms.append(p)
+            continue
+
+        # Otherwise, treat it as a name in the configuration file.
+        if not isinstance(p, str):
+            raise RuntimeError(f"Platform name {p} must be a string.")
+
+        # Explain the logic above in cases that look suspiciously like paths.
+        if "/" in p or os.path.splitext(p)[1] == ".json":
```

Comment:
Not necessary for now but just as a comment, `pathlib.Path` would make this a lot more readable using `p.suffix == "json"`

Reply:
Thanks, that's good to know.  There's quite a lot of `os.path` usage littered throughout the code, and I'd like us to be consistent.  I'll open an issue to track refactoring to use `pathlib` instead (where it makes sense).

Reply:
Opened as https://github.com/intel/code-base-investigator/issues/58.